API-GUIDELINE.md

```diff
@@ -0,0 +1,252 @@
+#API GUIDELINES
+
+##Overview
+These Guidelines aim to help developers to design APIs:
+
+ - simple
+ - consistent 
+ - self-explanatory
+ 
+The Keywords to have a good Guidelines are:
+ 
+ - Consistency => following the Guidelines
+ - Uniformity => checking in the code if there are other properties with similar name and adjust in base of it or improve it
+ - Explicability => explain with a few words the purpose of the property avoiding abbreviations (there are some *exceptions).
+
+##Scope
+The purpose of these guidelines is to achieve the definition of consistent practices and patterns for all API endpoints.
+
+##URI
+Our URIs are composed by:
+   
+    Base Uri like:
+        https://api.{region}.{cloudProvider}.commercetools.com 
+
+    {projectKey}
+
+    resources
+
+    by ID
+
+    by external ID with equals e.g. customers/key={key}
+
+See our documentation for more details https://docs.commercetools.com/http-api.html#hosts
+
+##Resources
+Resource is an object or representation of something, which has some associated data with it and can be helped by methods to operate on it.
+
+A resource can be an entity or a collection:
+ - Entity is a resource that has to be in singular.
+ - Collections are a set of resources that must be in plural.
+ 
+A resource may contain sub-collection resources.
+
+##Methods
+The HTTP verbs used are;
+
+ - GET -> retrieve data
+ - POST -> create or update data
+ - DELETE -> delete data
+
+It's important to keep in mind the concept of Idempotency. So for instance use GET to retrieve data and not to delete content!
```

Comment:
```suggestion
It's important to avoid unexpected behavior. So for instance use GET to retrieve data and not to delete content!
```

Idempotency is something different and mostly affects only update operations. Most of our updates are not idempotent. Cause either they produce errors or sending twice the same update doesn't result in the same response.